chemprop/data/splitting.py

```diff
@@ -216,38 +216,71 @@ def _unpack_astartes_result(
         train_idxs, val_idxs, test_idxs = result[-3], result[-2], result[-1]
     else:
         train_idxs, test_idxs = result[-2], result[-1]
-    return list(train_idxs), list(val_idxs), list(test_idxs)
+    return [list(train_idxs)], [list(val_idxs)], [list(test_idxs)]
 
 
 def split_component(
-    datapointss: Sequence[Sequence[MoleculeDatapoint | MulticomponentDatapoint]],
+    datapointss: Sequence[Union[Sequence[MoleculeDatapoint] | Sequence[ReactionDatapoint]]],
     split: SplitType | str = "random",
     key_index: int = 0,
     **kwargs,
 ):
     """Splits multicomponent data into training, validation, and test splits."""
 
     key_datapoints = datapointss[key_index]
-    train_idxs, val_idxs, test_idxs = split_data(key_datapoints, split=split, **kwargs)
+    train_idxss, val_idxss, test_idxss = split_data(key_datapoints, split=split, **kwargs)
+
+    train = [
+        [[datapoints[i] for i in train_idxs] for datapoints in datapointss]
+        for train_idxs in train_idxss
+    ]
+    val = [
+        [[datapoints[i] for i in val_idxs] for datapoints in datapointss] for val_idxs in val_idxss
+    ]
+    test = [
+        [[datapoints[i] for i in test_idxs] for datapoints in datapointss]
+        for test_idxs in test_idxss
+    ]
+
+    return train, val, test
 
-    match SplitType.get(split):
-        case SplitType.CV_NO_VAL | SplitType.CV:
-            # convert indices to datapoints for each fold
-            train = [
-                [[datapoints[i] for i in fold] for datapoints in datapointss] for fold in train_idxs
-            ]
-            val = [
-                [[datapoints[i] for i in fold] for datapoints in datapointss] for fold in val_idxs
-            ]
-            test = [
-                [[datapoints[i] for i in fold] for datapoints in datapointss] for fold in test_idxs
-            ]
-        case SplitType.SCAFFOLD_BALANCED | SplitType.RANDOM_WITH_REPEATED_SMILES | SplitType.RANDOM | SplitType.KENNARD_STONE | SplitType.KMEANS:
-            # convert indices to datapoints
-            train = [[datapoints[i] for i in train_idxs] for datapoints in datapointss]
-            val = [[datapoints[i] for i in val_idxs] for datapoints in datapointss]
-            test = [[datapoints[i] for i in test_idxs] for datapoints in datapointss]
-        case _:
-            raise RuntimeError("Unreachable code reached!")
 
+def splits_from_file(
+    datapointss: Sequence[Union[Sequence[MoleculeDatapoint] | Sequence[ReactionDatapoint]]],
+    splits_file: Path,
+):
```

Comment:
I took another look at the other tests in `test_data_utils.py` and see they all only test `split_data` (which returns indices) instead of `split_component`. I think the parallel case for `splits_from_file` is to test `parse_indices` so I have changed the test to just that. I don't think the other parts of `splits_from_file` (file reading, splitting) need tests because those aren't tested for `split_component`.